src/compiler/checker.ts

```diff
@@ -38841,11 +38841,17 @@ namespace ts {
             if (!checkGrammarDecoratorsAndModifiers(node) && hasEffectiveModifiers(node)) {
                 grammarErrorOnFirstToken(node, Diagnostics.An_export_assignment_cannot_have_modifiers);
             }
+
+            const typeTag = isInJSFile(node) ? getJSDocTypeTag(node) : undefined;

```

Comment:
getEffectiveTypeAnnotation instead of getJSDocTypeTag? It probably does the same thing now, but might be more flexible later.